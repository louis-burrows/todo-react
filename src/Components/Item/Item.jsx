import React from "react";
import Button from "../Button";
import styles from "./Item.module.scss";

const Item = (props) => {

  const {
    content, 
    deleteEntry, 
    index, 
    isCompleted, 
    toggleCompleted} = props;

  const completedStyle = isCompleted ? "isCompleted" : "notCompleted";

  return (
    <>
      <section className={styles.itemContainer}>
        <p className={styles[completedStyle]}>{content}</p>
        <Button text="Delete" handleClick={() => deleteEntry(index)} />
        <Button text="Mark as Done" handleClick={() => toggleCompleted(index, isCompleted)}/>
      </section>
    </>
  );
};

export default Item;
