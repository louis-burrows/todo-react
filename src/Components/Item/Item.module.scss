.itemContainer {
  max-width: 450px;
  flex-wrap: wrap;
  display: flex;
  flex-direction: column;
  margin: 30px auto;
  gap: 5px;
  font-weight: 600;
  p {
    text-align: left;
  }
  .notCompleted {
    color: rgb(44, 1, 1);
    font-size: 18px;
  }
  .isCompleted {
    color: rgb(26, 99, 26);
    text-shadow: 1px 1px 8px white;
  }
  .buttonSection {
    flex-direction: row;
    text-align: left;
    button {
      margin-right: 10px;
    }
  }
  
}

