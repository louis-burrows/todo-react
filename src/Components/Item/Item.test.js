import React from "react";
import { shallow } from "enzyme";
import Item from "./Item";

describe("Item tests", () => {

  let testComponent;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn();
    testComponent = shallow(<Item />);
  })

  it("should render", () => {
    expect(testComponent).toBeTruthy();
  });

  // it('should call the parent function with click', () => {
  //   testComponent.find('.buttonToggle').simulate('click');
  //   expect(mockFn).toHaveBeenCalledWith();
  // })

});

//  DOES THE PARAGRAPH DISPLAY THE PROP OF content?
//  Provide a prop of text content to the component.
//  Expect the content of the paragraph tag to contain the text from the delivered prop.


//  DOES THE DELETE BUTTON CALL THE FUNCTION PASSED AS PROPS?
//  Create a mock function.
//  Deliver the mock function as a prop to the delete button in the manner that it would receive it's function.
//  Simulate a click of the delete button.
//  Expect the mock function to run.


//  DOES THE TOGGLE BUTTON CALL THE FUNCTION PASSED AS PROPS?
//  Create a mock function.
//  Deliver the mock function as a prop to the toggle button in the manner that it would receive it's function.
//  Simulate a click of the delete button.
//  Expect the mock function to run.


//  IS TOGGLING THE completed CLASS OF THE PARAGRAPH REVERSIBLE, DOES THE CLASSNAME REVERT ON EVERY CLICK?
//  Expect the className of the paragraph to be notCompleted.
//  Simulate a click of the toggle button.
//  Expect the className of the paragraph to be isCompleted.
//  Simulate a click of the toggle button.
//  Expect the className of the paragraph to be notCompleted.


//  ARE THE ELEMENTS RENDERED IN THE STATES THEY SHOULD BE PRIOR TO ANY USER INPUT?
//  On the instance of a new component, expect p.className.notCompleted to be true.
//  Expect the paragraph and both buttons to be rendered.
