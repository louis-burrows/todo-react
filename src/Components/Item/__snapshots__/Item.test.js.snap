// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item tests should match the snapshot 1`] = `ShallowWrapper {}`;
